File: numbers/bindec.lua
binary = {0, 0, 0};
power, base = 0, 2;
decimal = 0;

for bindex in pairs(binary) do
	io.write("input a number: ")
	b = io.read("n")
	binary[bindex] = b
end

for bindex=#binary, 0, -1 do
	if binary[bindex] == nil then
		break	
	end
	power_result = (base^power) * binary[bindex]
	decimal = decimal + power_result
	power = power + 1
end
 
print("decimal is " .. decimal)

File: numbers/bindec.c
#include <stdio.h>
#include <math.h>


int main() {

	int binArr[3] = {0, 0, 0};
	int binArrLength = sizeof(binArr) / sizeof(binArr[0]);
	int i;
	int temp;
	int power = 0;
	double base = 2.0;
	double decimal = 0;
	double powerResult;

	for (i=0; i<binArrLength; i++) {
		printf("input a number: ");
		scanf("\n%d", &temp);
		binArr[i] = temp;	
	}

	for (i=binArrLength-1; i>-1; i--) {
		powerResult = pow(base, power) * binArr[i];
		decimal = decimal + powerResult;
		power++;
	}

	printf("decimal is: %.f\n", decimal);

	return 0;
}
